core/Updater/Migration/Db/ChangeColumnTypes.php

```diff
@@ -0,0 +1,28 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+namespace Piwik\Updater\Migration\Db;
+
+/**
+ * @see Factory::changeColumnTypes()
+ * @ignore
+ */
+class ChangeColumnTypes extends Sql
+{
+    public function __construct($table, $columns)
+    {
+        $changes = array();
+        foreach ($columns as $columnName => $columnType) {
+            $changes[] = sprintf("CHANGE `%s` `%s` %s", $columnName, $columnName, $columnType);
+        }
+
+        $sql = sprintf("ALTER TABLE `%s` %s", $table, implode(', ', $changes));
+
+        parent::__construct($sql, static::ERROR_CODE_DUPLICATE_COLUMN);
```

Comment:
i'm changing it to unknown and will see what happens. It does definitely make sense. Probably 1060 was used once to rename a column and then copied for all further `change column` uses.